### ISSUE TITLE
Unable to use KBFS for own account. Can see other's public folders.

### ISSUE THREAD

**Issue by jaybosamiya** [keybase/client#5826]
My account is https://keybase.io/jaybosamiya

Even though I am logged in, it says "jaybosamiya does not have read access to /keybase/private/jaybosamiya" when I try `cd /keybase/private/jaybosamiya`.

Additionally, for the public directory, it says `ls: cannot open directory .: Input/output error` when I try to `ls` it, though I am able to `cd` to it.

This is a recent issue, since I was able to successfully access and use these until very recently.

For further information: I am still able to access other people's public files using `cd` and `ls` (for example `/keybase/public/chris` works). It is only my own account that doesn't seem to work.

**Comment by cjb** [comment]
Thanks for the report, CC @strib.

Please could you run `keybase log send` and tell us the ID?

**Comment by jaybosamiya** [comment]
Log ID: `fce1ce9cb3fba8dd3822781c`

**Comment by strib** [comment]
cc: @maxtaco @patrickxb 

Well, this is a new one.  You do appear to be properly logged in.  Strangely, your requests for your public folder, which definitely already exists and which I and keybase.pub can read just fine, is getting this error from the server:

```
82420 2017-02-11T17:55:12.964964 ▶ [DEBU kbfs md_ops.go:254] 2e5 GetForHandle done with err=Bad MD server request: must provide either uid or username (error 100) [tags:FID=svRTVbzAUZBo6g6jvY95lw]
82421 2017-02-11T17:55:12.965038 ▶ [DEBU kbfs kbfs_ops.go:435] 2e6 Done: libkbfs.MDServerErrorBadRequest{Reason:"must provide either uid or username (error 100)"} [tags:FID=svRTVbzAUZBo6g6jvY95lw]
```

I _think_ this error is returned when we call `auth.CredentialAuthority.CheckUssrs`, but I can't actually find where the error is returned.  It probably means the UID is 0 being checked is 0, or somehow bad.

Furthermore, access to the private folder fails with:

```
2017-02-11T17:57:23.162536 ▶ [DEBU FUSE serve.go:845] 6c1 -> [ID=0x229] Lookup error=permission denied: jaybosamiya does not have read access to directory /keybase/private/jaybosamiya
```

Which is obviously weird, and seems like the UIDs we're trying to check don't match.

@maxtaco @patrickxb: it feels to me like `Resolve2(jaybosamiya)` is returning a 0 UID, or some other bad UID.

In the service log, this looks somewhat interesting:

```
2017-02-11T17:55:05.568345 ▶ [DEBU keybase standard.go:146] 6ec + IdentifyHandler#Resolve2(jaybosamiya) [tags:RSLV=Vg1QJ7Fc-voi]
2017-02-11T17:55:05.568380 ▶ [DEBU keybase standard.go:146] 6ed + IdentifyHandler#Resolve2(jaybosamiya) [tags:RSLV=6OOyIwgSSKqF]
2017-02-11T17:55:05.568599 ▶ [DEBU keybase standard.go:146] 6ee - IdentifyHandler#Resolve2(jaybosamiya) -> ok [tags:RSLV=Vg1QJ7Fc-voi]
2017-02-11T17:55:05.568665 ▶ [DEBU keybase standard.go:146] 6ef - IdentifyHandler#Resolve2(jaybosamiya) -> ok [tags:RSLV=6OOyIwgSSKqF]
```

This looks like two Resolve2 requests happened concurrently.  Is it possible this messed up a cache somehow and left the UID for the user in a bad state?

**Comment by maxtaco** [comment]
Thanks @strib for taking a look. I made changes here last week, so it could be I introduced a bug.  I need a few minutes to stare at it, probably can't get to it until later tonight.

**Comment by maxtaco** [comment]
@jaybosamiya  in terms of short-term workarounds, you can `keybase db nuke` and then restart your keybase (via `run_keybase`) and it should work.

**Comment by jaybosamiya** [comment]
Thanks :) It does seem like the db was left in a bad state somehow, and the db nuke worked. I am unable to reproduce the error after this nuke though. However, if I do find a way to reproduce the bug, I will comment in this thread.

**Comment by maxtaco** [comment]
Cool, well that lends credence to @strib's theory above.  Glad it fixed buggy situation for now.  I'll work on a longer term fix.

**Comment by maxtaco** [comment]
In some bad timing, I just removed a bunch of debugging along this code path because it was spammy and I thought this code was bug-free.   Anywho, by inference, it seems the problem was present earlier in the trace:

```
017-02-11T00:50:25.954592 ▶ [DEBU keybase engine.go:65] 218f + RunEngine(ResolveThenIdentify2)
2017-02-11T00:50:25.954626 ▶ [DEBU keybase standard.go:146] 2190 + ResolveThenIdentify2#Run [tags:ID2=FMyXwfvU552Q]
2017-02-11T00:50:25.954814 ▶ [DEBU keybase standard.go:146] 2191 + RunEngine(Identify2WithUID) [tags:ID2=FMyXwfvU552Q]
2017-02-11T00:50:25.954852 ▶ [DEBU keybase standard.go:146] 2192 + Identify2WithUID#Run(UID=, Assertion=jaybosamiya) [tags:ID2=GCysLoSkdRb3]
2017-02-11T00:50:25.954869 ▶ [DEBU keybase standard.go:146] 2193 | Full Arg: &{SessionID:264 Uid: UserAssertion:jaybosamiya Reason:{Type:ID Reason:Profile Resource:} UseDelegateUI:false AlwaysBlock:false NoErrorOnTrackFailure:true ForceRemoteCheck:false NeedProofSet:true AllowEmptySelfID:true NoSkipSelf:true CanSuppressUI:false IdentifyBehavior:DEFAULT_KBFS ForceDisplay:false} [tags:ID2=GCysLoSkdRb3]
2017-02-11T00:50:25.954898 ▶ [DEBU keybase standard.go:146] 2194 - Identify2WithUID#Run(UID=, Assertion=jaybosamiya) -> No UID given but one was expected [time=29.624µs] [tags:ID2=GCysLoSkdRb3]
2017-02-11T00:50:25.954938 ▶ [DEBU keybase standard.go:146] 2195 - RunEngine(Identify2WithUID) -> ERROR: No UID given but one was expected [tags:ID2=GCysLoSkdRb3]
```

This seems to indicate that we pulled a 0-UID out of  a cache somewhere....

**Comment by maxtaco** [comment]
So @strib therefore I don't think this corruption was due to the overlapping requests, since the log shows problems before then.  I spent a bunch of time staring at the logs and don't have a great idea.  I might resolve (hah) to throw in some more logging, error-checking, then wait for it to happen again.

**Comment by maxtaco** [comment]
I think I see the problem. Tracing the log all the way back (back when we still had way more debugging here), there is evidence that a resolution for `jaybosamiya` failed on Jan 31. Back then, there was a bug in the resolver that cached errors permanently to disk.  So if there was any sort of network error, then a bum resolution would wind up permanently cached.  That was fixed in e5f088f7ac393663dddb77531d9e68907da82f09.  So I think bug shouldn't be an issue going forward.  But there might be people in the wild with bad local DBs as a result of this bug.
